src/main/osd/osd_elements.c

```diff
@@ -1144,6 +1144,39 @@ static void osdElementEfficiency(osdElementParms_t *element)
         tfp_sprintf(element->buff, "----%c/%c", SYM_MAH, unitSymbol);
     }
 }
+
+int16_t osdGetBoundaryFromDisplay(uint8_t fov, int16_t error)
+{
+    const uint8_t fov2 = fov / 2;
+    return constrain(error, -fov2, fov2-1);
+}
+
+static void osdElementProjectHomePoint(osdElementParms_t *element)
+{
+    if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
+        const int16_t center_x = osdConfig()->canvas_cols / 2;
+        const int16_t center_y = osdConfig()->canvas_rows / 2;
+        int16_t error_x = DECIDEGREES_TO_DEGREES(GPS_directionToHome - attitude.values.yaw);
```

Comment:
Another way to calculate home symbol position is to get home position in earth frame, rotate it to body frame using IMU functions and then project it to camera plane (with screen bounds checking). 
It will handle roll/pitch automatically, avoid trig functions and overall be cleaner. But I'm not that sure how will it work in practice.

Reply:
Mhh you mean to get GPS to earth frame, But then you need Information of earth radius and so on I dont know if there comes more error insider calculation resuld!?

Reply:
Earth frame in IMU sense (ground frame, but EF is used in IMU).
```
homePositionEF = {gps.x - home.x, gps.y - home.y, -altitude);
imuTransformVectorEarthToBody(homePositionBF, homePositionEF);

// IIRC, x is forward in body frame
// TODO: handle  homePositionBF.x == 0 division
cameraProjection = {.x = homePositionBF.y / homePositionBF.x, homePositionBF.z /  homePositionBF.x };

ODS_rectangle = {.x = +-sin(fov_h/2), .y=+-sin(fov_v/2);
if (homePositionBF.x > 0 && cameraProjection.[x,y] in ODS_rectangle) {
   mapScren(cameraProjection);
} else {
   // TODO: project  cameraProjection to screen boundary
   // maybe use second symbol to display 'home is behind us' and use projection as for homePositionBF.x > 0
}

Reply:
Maybe you van implementiert it with a switch to my pr? So then we van compare it by switching and take the Vetter solution hat Do you mean?

Reply:
I have take a look. I only find this function:
https://github.com/betaflight/betaflight/blob/9f816b678d3c99462a5bbbeed70afb65fd99969a/src/main/flight/imu.c#L711C6-L711C53

also for homePositionEF I find something like this, where we can get the values?!
`GPS_distance_cm_bearing(&gpsSol.llh.lat, &gpsSol.llh.lon, &GPS_home[GPS_LATITUDE], &GPS_home[GPS_LONGITUDE], &dist, &dir);`

I do not realy know how to implement your solution/Idea.